rpc/codec/conn.go

```diff
@@ -5,92 +5,85 @@
 package codec
 
 import (
+	"bufio"
 	"encoding/binary"
-	"errors"
 	"io"
 	"net"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
 )
 
+type frameBuf struct {
+	size [binary.MaxVarintLen64]byte
+}
+
+var framePool = sync.Pool{
+	New: func() interface{} {
+		return &frameBuf{}
+	},
+}
+
 func sendFrame(w io.Writer, data []byte) (err error) {
 	// Allocate enough space for the biggest uvarint
-	var size [binary.MaxVarintLen64]byte
+	buf := framePool.Get().(*frameBuf)
```

Comment:
I found this using the memory profiler: `go test -bench=. -memprofile=prof -memprofilerate=1`. The `-memprofilerate=1` setting tells the memory profiler to record every allocation. This slows the benchmark down a lot, but gives a price view of where the allocations are occurring. I then use `go tool pprof --alloc_objects <binary> prof` to view the output.